worker-queue-rabbit/src/main/java/com/github/workerframework/queues/rabbit/WorkerQueueConsumerImpl.java

```diff
@@ -204,21 +216,37 @@ private void handleTaskDataInjection(final TaskMessage taskMessage, final long i
     }
 
     private byte[] retrieveTaskDataFromStore(final String taskMessageStorageRef, final long inboundMessageId) 
-            throws InvalidDeliveryException, TransientDeliveryException
+            throws ReferenceNotFoundException, TransientDeliveryException
     {
         try (final var inputStream = dataStore.retrieve(taskMessageStorageRef)) {
             final var taskData = inputStream.readAllBytes();
             offloadedPayloadsToDelete.put(inboundMessageId, taskMessageStorageRef);
             return taskData;
-        } catch (final ReferenceNotFoundException ex) {
-            throw new InvalidDeliveryException("TaskMessage's TaskData could not be retrieved from DataStore", 
-                    inboundMessageId, ex);
         } catch (final IOException | DataStoreException ex) {
+            if (ex instanceof ReferenceNotFoundException) {
+                throw (ReferenceNotFoundException)ex;
+            }
             throw new TransientDeliveryException(
                 "TaskMessage's TaskData could not be retrieved from DataStore", inboundMessageId, ex);
         }
     }
 
+    private void sendTrackingReport(final RabbitTaskInformation taskInformation, final TaskMessage taskMessage,
```

Comment:
Isn't this only used to send a failure tracking report?
```suggestion
    private void sendFailureTrackingReport(final RabbitTaskInformation taskInformation, final TaskMessage taskMessage,
```